supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesInitializationActor.scala

```diff
@@ -52,22 +61,41 @@ class JesInitializationActor(override val workflowDescriptor: BackendWorkflowDes
   /**
     * Abort all initializations.
     */
+  // TODO PBE ??? is not cool
   override def abortInitialization(): Unit = ???
 
-  //TODO: Workflow options may need to be validated for JES.
+  //TODO PBE: Workflow options may need to be validated for JES.
 
   /**
     * A call which happens before anything else runs
     */
-  override def beforeAll(): Future[Unit] = {
-    publishWorkflowRoot(workflowPaths.workflowRootPath.toString)
-    if (jesConfiguration.needAuthFileUpload) writeAuthenticationFile()
-    else Future.successful(())
+  override def beforeAll(): Future[Option[BackendInitializationData]] = {
+
+    def buildGenomics: Genomics = {
+      val jesAttributes = jesConfiguration.jesAttributes
+      GenomicsFactory(
+        jesConfiguration.googleConfig, jesAttributes.genomicsAuth.credential(workflowDescriptor.workflowOptions.toGoogleAuthOptions), jesAttributes.endpointUrl)
+    }
+
+    def buildGcsFileSystem: GcsFileSystem = {
+      val storage = jesConfiguration.jesAttributes.gcsFilesystemAuth.buildStorage(
+        workflowDescriptor.workflowOptions.toGoogleAuthOptions, jesConfiguration.googleConfig)
+      GcsFileSystem(GcsFileSystemProvider(storage))
+    }
+
+    for {
+      // generate single filesystem and genomics instances
+      gcsFileSystem <- Future(buildGcsFileSystem)
```

Comment:
Why not set these two Futures up outside of the forcomp since they don't seem to rely on each other?